File: Digital Marketing/replace_icons.py
import os
import glob

# FontAwesome link
fa_link = '  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css" />\n</head>'

replacements = {
    '📘': '<i class="fa-brands fa-facebook" style="color: #1877F2;"></i>',
    '🐦': '<i class="fa-brands fa-twitter" style="color: #1DA1F2;"></i>',
    '💼': '<i class="fa-brands fa-linkedin" style="color: #0A66C2;"></i>',
    '📸': '<i class="fa-brands fa-instagram" style="color: #E1306C;"></i>',
    '▶️': '<i class="fa-brands fa-youtube" style="color: #FF0000;"></i>',
    '🎵': '<i class="fa-brands fa-pinterest" style="color: #E60023;"></i>',
    '</head>': fa_link
}

html_files = glob.glob('d:/SMM/SMM 2/*.html')
for file_path in html_files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if 'font-awesome' in content:
        replacements.pop('</head>', None)
        
    for text, replacement in replacements.items():
        content = content.replace(text, replacement)
        
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
        
print("Successfully replaced emojis with original logos in all HTML files.")

File: Digital Marketing/replace_icons.js
const fs = require('fs');
const path = require('path');
const dir = 'd:/SMM/SMM 2';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html'));
files.forEach(file => {
  let content = fs.readFileSync(path.join(dir, file), 'utf8');

  // Insert FontAwesome before </head> if not present
  if (!content.includes('font-awesome')) {
    content = content.replace('</head>', '  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css" />\n</head>');
  }

  // Replace emojis with FontAwesome brand icons + original colors
  content = content.replace(/📘/g, '<i class="fa-brands fa-facebook" style="color: #1877F2;"></i>');
  content = content.replace(/🐦/g, '<i class="fa-brands fa-twitter" style="color: #1DA1F2;"></i>');
  content = content.replace(/💼/g, '<i class="fa-brands fa-linkedin" style="color: #0A66C2;"></i>');
  content = content.replace(/📸/g, '<i class="fa-brands fa-instagram" style="color: #E1306C;"></i>');
  content = content.replace(/▶️/g, '<i class="fa-brands fa-youtube" style="color: #FF0000;"></i>');
  content = content.replace(/🎵/g, '<i class="fa-brands fa-pinterest" style="color: #E60023;"></i>');

  fs.writeFileSync(path.join(dir, file), content, 'utf8');
});
console.log('Icons and FontAwesome added to ' + files.length + ' files.');
